provider/pkg/schema/docs/content/cfn-custom-resource.md

```diff
@@ -0,0 +1,56 @@
+The Custom Resource Emulator allows you to use AWS CloudFormation Custom Resources directly in your Pulumi programs. It provides a way to invoke AWS Lambda functions that implement custom provisioning logic following the CloudFormation Custom Resource protocol.
+
+> **Note**: Currently, only Lambda-backed Custom Resources are supported. SNS-backed Custom Resources are not supported at this time.
+
+## Example Usage
+
+```typescript
+import * as aws from "@pulumi/aws-native";
+
+const bucket = new aws.s3.Bucket('custom-resource-emulator');
+
+// Create a Custom Resource that invokes a Lambda function
+const cr = new aws.cloudformation.CustomResourceEmulator('cr', {
+    bucketName: bucket.id,
+    bucketKeyPrefix: 'custom-resource-emulator',
+    customResourceProperties: {
+        hello: "world"
+    },
+    serviceToken: "arn:aws:lambda:us-west-2:123456789012:function:my-custom-resource",
```

Comment:
This lambda would be provisioned by CF, is that right?

Reply:
No, it is provisioned by the user in their Pulumi program or CloudFormation Stack.

Reply:
This Example usage should probably have a link to how to do that.

Reply:
The `Lambda Function Requirements` section explains how to do that. I don't think we should add a full Custom Resource implementation here. That would essentially be the `cfn-custom-resource` example I've added (and that's multiple files)

Reply:
Yeah, I can add that once it's merged into master :)